index.html

```diff
@@ -6,7 +6,7 @@
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <link rel="stylesheet" href="styles.css">
   <script src="data.js"></script>
-  <script src="index.js"></script>
+  <script src="index.js" defer></script>
```

Comment:
This is my preferred way of making sure the DOM is fully loaded before the Javascript code is executed. Other options would be to move the script tag to the bottom of the body tag or adding an DOMContentLoaded event listener 💯